hamster_gtk/misc/widgets/raw_fact_entry.py

```diff
@@ -0,0 +1,127 @@
+
+    # Callbacks
+    def _on_facts_changed(self, evtl):
+        """Callback triggered when facts have changed."""
+        self.set_completion(RawFactCompletion(self._controller))
+
+
+class RawFactCompletion(Gtk.EntryCompletion):
+    """
+    Return a completion instance to match 'activity@category' strings.
+
+    Returns:
+        Gtk.EntryCompletion: Completion instance.
+    """
+
+    def __init__(self, controller, *args, **kwargs):
+        """instantiate class."""
+        super(RawFactCompletion, self).__init__(*args, **kwargs)
+        self._controller = controller
+        self.set_model(self._get_store())
+        self.set_text_column(0)
+        self.set_match_func(self._match_anywhere, None)
+        self.connect('match-selected', self._on_match_selected)
+
+    def _get_store(self):
+        store = Gtk.ListStore(GObject.TYPE_STRING, GObject.TYPE_STRING, GObject.TYPE_STRING)
+        for activity in self._get_activities():
+            category = ''
+            if activity.category:
+                category = text_type(activity.category.name)
+            store.append([
+                helpers.serialise_activity(activity), text_type(activity.name), category
+            ])
+        return store
+
+    def _get_activities(self):
+        today = datetime.date.today()
+        recent_activities = [fact.activity for fact in self._controller.facts.get_all(
+            start=today, end=today)]
+        return OrderedSet(recent_activities)
+
+    def _match_anywhere(self, entrystr, iter, data, *args):
+        """
+        Check if the entrysring is a substring of our text_column.
+
+        Args:
+            entrystr (str): The text extracted from the entry so far. Note that key is
+                normalized and case-folded (see GLib.utf8_normalize() and
+                GLib.utf8_casefold()).
+            iter: Iterator position indicating the model row we check against.
+            data: Arbitrary user data.
+
+        Returns:
+            boot: ``True`` if ``entrystring`` is a substring, ``False`` if not.
+
+        Note this is a custom match function, for details on the general generic
+        solution [please see|https://lazka.github.io/pgi-docs/#Gtk-3.0/
+        callbacks.html#Gtk.EntryCompletionMatchFunc].
+        """
+        # 'activity@category' string that we try to match against.
+        modelstr = _u(self.get_model()[iter][0])
+        # Parse the entire entry string to extract all semantic information.
+        fact = Fact.create_from_raw_fact(_u(entrystr))
+        # Make the extracted ``Fact`` available to other callbacks.
+        self._tmp_fact = fact
+        # If the parsed string seems to contain any activity related
+        # information, encode it in its canonical form.
+        if fact.activity:
+            activity_string = helpers.serialise_activity(fact.activity)
+        return activity_string in modelstr
+
+    def _on_match_selected(self, model, iter):
```

Comment:
Similarly to https://github.com/projecthamster/hamster-gtk/pull/138#discussion_r87098991 there should be `completion` parameter (see https://lazka.github.io/pgi-docs/#Gtk-3.0/classes/EntryCompletion.html#Gtk.EntryCompletion.signals.match_selected). Otherwise this error is raised:


```python
TypeError: _on_match_selected() takes 3 positional arguments but 4 were given
```